client-misk/src/main/kotlin/app/cash/backfila/client/misk/client/BackfilaClientLoggingSetup.kt

```diff
@@ -0,0 +1,71 @@
+package app.cash.backfila.client.misk.client
+
+import javax.inject.Inject
+import misk.logging.getLogger
+import org.slf4j.MDC
+
+interface BackfilaClientLoggingSetupProvider {
+  fun <T> withBackfillRunLogging(backfillName: String, backfillId: String, wrapped: () -> T): T
+  fun <T> withBackfillPartitionLogging(backfillName: String, backfillId: String, partitionName: String, wrapped: () -> T): T
+}
+
+class BackfilaClientNoLoggingSetupProvider
+@Inject constructor() : BackfilaClientLoggingSetupProvider {
+  override fun <T> withBackfillRunLogging(backfillName: String, backfillId: String, wrapped: () -> T): T {
+    return wrapped.invoke()
+  }
+
+  override fun <T> withBackfillPartitionLogging(backfillName: String, backfillId: String, partitionName: String, wrapped: () -> T): T {
+    return wrapped.invoke()
+  }
+}
+
+class BackfilaClientMDCLoggingSetupProvider
+@Inject constructor() : BackfilaClientLoggingSetupProvider {
+
+  override fun <T> withBackfillRunLogging(backfillName: String, backfillId: String, wrapped: () -> T): T {
```

Comment:
would it be better to have one method with nullable `partitionName`?

Reply:
Yeah I dunno. I think for the same reason about different implementations for the server and client I kept them different.